### DISCUSSION TITLE
CnC ZH determine the winner based on the replay file

### DISCUSSION THREAD

**Discussion by F4R4N** [OpenSAGE/OpenSAGE#901] [Q&A]
Hi,
I am interested in determining the winner of a General ZH match based on the replay file. But it seems like not much information about “.rep” files is available. Can you share any experiences that might help me on my journey? 
I appreciate any help you can provide.
Faran

**Comment by charliefoxtwo** [comment]
You can check out the data structure we use to parse and store replay files [here](https://github.com/OpenSAGE/OpenSAGE/tree/master/src/OpenSage.Game/Data/Rep), which may be the best place to start.

To my knowledge replays largely just store actions, which are then _replayed_ (go figure) in the current game engine. I'm not certain if the winner can be determined from a replay file.
